content/en/developers/integrations/log_pipeline.md

```diff
@@ -16,111 +16,165 @@ further_reading:
+
+Map the log `service` tag to the name of the [service][26] producing telemetry
+: Use the [Service Remapper][7] to remap the `service` attribute. When the source and service are not different services, set `service` to the same value as the `source` tag. `service` tags must be in lowercase. 
+
+Map the log's internal timestamp to it's official Datadog timestamp
+: Use the [Date Remapper][4] to define the official timestamp for logs. If the log timestamp does not already map to a [standard date attribute][28] it defaults to the time of ingestion at Datadog's endpoint.
+
+Map the custom status attributes of the logs to the official Datadog `status` attribute
+: Use a [Status Remapper][25] to remap the `status` of a log, or a [Category Processor][19] for statuses mapped to a range (as with HTTP status codes).
 
-**Requirements:**
+Map the custom message attribute of the logs to the official Datadog `message` attribute
+: Use the [message remapper][9] to define the official message of the log if application logs do not map to the standard message attribute. This allows users to search for logs using free text.
 
-- Use the [date remapper][4] to define the official timestamp for logs.
-- Use a status remapper to remap the `status` of a log, or a [category processor][19] for statuses mapped to a range (as with HTTP status codes).
-- Use the attribute [remapper][5] to remap attribute keys to standard [Datadog attributes][6]. For example, an attribute key that contains the client IP must be remapped to `network.client.ip` so Datadog can display Technology Partner logs in out-of-the-box dashboards. Remove original attributes when remapping by using `preserveSource:false` to avoid duplicates.
-- Use the [service remapper][7] to remap the `service` attribute or set it to the same value as the `source` attribute.
-- Use the [grok processor][8] to extract values in the logs for better searching and analytics. To maintain optimal performance, the grok parser must be specific. Avoid wildcard matches.
-- Use the [message remapper][9] to define the official message of the log and make certain attributes searchable by full text.
+Set a namespace for custom attributes within your logs
+: Generic log attributes that do not map to a [Datadog Standard Attribute][6] must be namespaced if they are mapped to [Facets][14].
+Use the [Attribute Remapper][5] to set attribute keys to a new namespaced attribute. 
 
-For a list of all log processors, see [Processors][10].
+1. Expand the newly created pipeline and click the "Add Processor" button to begin building your pipeline using processors.
+2. If the integration's logs aren't `JSON` formatted, extract attribute information using the [Grok Processor][8]. Use grok processors to parse out attributes and enrich logs prior to remapping or further processing.
```

Comment:
can we flag this one as optional?

Reply:
I think it's best we keep that as a direct instruction due to how frequently people might skip optional instructions.